### PR TITLE
Move observation configuration to program cache

### PR DESCRIPTION
**Pull request by rpiaggio**
You can consider this PR as the final part of the program cache refactor. There was an intermediate state where we were doubly querying parts of the observation. Now everything's in the cache, including the configuration, and we query only once.

Switching observations in the observations tab should be faster than before.

Also, as a "side effect", the configuration undo stacks become unified with the general observation stacks.

### CODE REVIEW

**Comment by toddburnside** [comment]
> Also, as a "side effect", the configuration undo stacks become unified with the general observation stacks.

Inching towards universal undo. 🤣

File: explore/src/main/scala/explore/tabs/ObsTabTiles.scala [hunk 1]
```diff
@@ -230,39 +249,36 @@ object ObsTabTiles:
...
-          val (obsViewPot, _, observingMode, scienceReqs, chartExposureTime) =
-            obsProperties(obsView)
+          // PREVIOUS CODE WAS: (BUT WE ALWAYS HAD "none" IN itcExposureTime
+          //   val chartExposureTime                        = obsView.toOption
+          //     .flatMap(
+          //       _.get.itcExposureTime
+          //         .map(r => ItcChartExposureTime(OverridenExposureTime.FromItc, r.time, r.count))
+          //     )
+          //     .orElse(obsView.toOption.flatMap(_.get.itc.map(_.toItcChartExposureTime)))
+          //   (obsViewPot, scienceData, observingMode, scienceReqs, chartExposureTime)
```
Comment (toddburnside) [inline explore/src/main/scala/explore/tabs/ObsTabTiles.scala:259]: Do you want to keep this commented code?

**Review by toddburnside — APPROVED** [review]
LGTM - It's nice to have this all wrapped up! :)

File: explore/src/main/scala/explore/tabs/ObsTabTiles.scala (hunk 1 above)
Comment (rpiaggio) [reply explore/src/main/scala/explore/tabs/ObsTabTiles.scala:259]: I think yes, for the moment. This is a pending fix. WDYT @cquiroz ?

**Pull request merged**